webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ArchiveApi.scala

```diff
@@ -1054,6 +1055,50 @@ trait SaveArchiveService {
   def save(archive: PolicyArchive, mergePolicy: MergePolicy, actor: EventActor): IOResult[Unit]
 }
 
+final case class TechniqueArchiveMod(added: Chunk[ResourceFile], updated: Chunk[ResourceFile])
+
+object SaveArchiveServicebyRepo {
+
+  /*
+   * Create the resource diff between the archive technique and the corresponding path
+   * (assumed to be the technique base directory, ie ../category/1.0/techniqueId).
+   * If file does not exists, everything is marked as "new", including the base directory.
+   */
+  def buildDiff(t: TechniqueArchive, techniqueBaseDirectory: File): IOResult[Chunk[ResourceFile]] = {
+    for {
+      // path ot 't' are relative to technique. Files we need to add as new in the end
+      addedRef <- Ref.make(Chunk.fromIterable(t.files.map(_._1)))
+      diffRef  <- Ref.make(Chunk[ResourceFile]())
+      // get all file path (relative to technique dir)
+      existing <- IOResult.attempt {
+                    if (techniqueBaseDirectory.exists) {
+                      techniqueBaseDirectory
+                        .collectChildren(!_.isDirectory) // we don't have directories in the archive, only files
+                        .toList
+                        .map(_.pathAsString.replaceFirst("\\Q" + techniqueBaseDirectory.pathAsString + "/\\E", ""))
+                    } else { // technique or technique version does not exists
+                      Chunk.empty
+                    }
+                  }
```

Comment:
The inferred type is a bit weird because in the first conditional branch the type is `List[String]` and in the second one a `zio.Chunk`, so maybe an empty List should be there ? In the end it is used as an iterable so even keeping an iterable would do...

Reply:
```suggestion
      existing <- IOResult.attempt {
                    if (techniqueBaseDirectory.exists) {
                      techniqueBaseDirectory
                        .collectChildren(!_.isDirectory) // we don't have directories in the archive, only files
                        .map(_.pathAsString.replaceFirst("\\Q" + techniqueBaseDirectory.pathAsString + "/\\E", ""))
                        .to(Iterable)
                    } else { // technique or technique version does not exists
                      Iterable.empty[String]
                    }
                  }
```

Reply:
I realized that it is code that has been moved, but I think it should be changed because it slows the IDE because of the huge type being inferred `Seq[String] with StrictOptimizedSeqOps[String,[_]Seq[_] with immutable.StrictOptimizedSeqOps[_,[_]Seq[_] with immutable.StrictOptimizedSeqOps[_,[_]Seq[_] with Serializable,Seq[_] with Serializable] with Serializable,Seq[_] with immutable.StrictOptimizedSeqOps[_,[_]Seq[_] with Serializable,Seq[_] with Serializable] with Serializable] with Serializable{def iterableFactory: scala.collection.SeqFactory[[A]Seq[A] with Serializable]},Seq[String] with StrictOptimizedSeqOps[String,[_]Seq[_] with immutable.StrictOptimizedSeqOps[_,[_]Seq[_] with Serializable,Seq[_] with Serializable] with Serializable,Seq[String] with StrictOptimizedSeqOps[String,[_]Seq[_] with Serializable,Seq[String] with Serializable] with Serializable] with Serializable{def iterableFactory: scala.collection.SeqFactory[[A]Seq[A] with java.io.Serializable]}] with Serializable{def iterableFactory: scala.collection.SeqFactory[[A]Seq[A] with scala.collection.immutable.StrictOptimizedSeqOps[A,[_]Seq[_] with java.io.Serializable,Seq[A] with java.io.Serializable] with java.io.Serializable]}` :sweat_smile: